compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -758,6 +762,8 @@ object Scanners {
         if endMarkerTokens.contains(lookahead.token)
           && source.offsetToLine(lookahead.offset) == endLine
         then
+          if rewriteToIndent && lookahead.token == MATCH then
+            patch(Span(offset, offset + 3), "`end`")
```

Comment:
`end match {` must be rewritten to ````end` match```, otherwise `end match` is scanned as an end marker.